### ISSUE TITLE
[Feature Request] masking time steps with FixedNoiseMultiTaskGP

### ISSUE THREAD

**Issue by JFagin** [pytorch/botorch#1637]
Is there support for masking? I have a multivariate function with known observations at irregularly spaced sampling. I would like to use FixedNoiseMultiTaskGP since the noise is known but the time vectors of each observation are not the same length for each task. I would like to use one evenly spaced time vector and mask out which tasks are observed at each point and with what uncertainty.

**Comment by Balandat** [comment]
Hmm not sure I fully understand - Do you want to build a multivariate time series model here? Or are you not interested in predicting across time and want to do some kind of averaging across time since the you don't really expect a time dependency and samples are iid w.r.t. time?

**Comment by JFagin** [comment]
Thanks for the reply. Sorry if I didn't make things clear, but essentially I have a problem like:

x1 = torch.rand(50)
y1 = torch.sin(X1)
y_var1 = 0.1*torch.rand(y1.shape)

x2 = torch.rand(25)
y2 = torch.sin(X2)
y_var2 = 0.1*torch.rand(y2.shape)

Where I want to fit a multi task model jointly between y1 and y2 since their outputs are correlated. Is this possible when y1 and y2 do not have the same shape. My thought was to use some kind of masking so I have a single time vector like:

train_x = torch.linspace(0, 1, 100)
train_y = torch.stack([torch.sin(train_x),torch.cos(train_x)],-1)
train_y[10:20,0] = 0.0 # some points are unobserved in dim 1
train_y[50:60,1] = 0.0 # some points are unobserved in dim 2
y_var = 0.1*torch.rand(train_y.shape)
mask = (train_y != 0.0).type_as(train_y)  # 1 when observed, 0 when not observed
train_y += torch.normal(mean=0,std=y_var.sqrt()) # Add Gaussian noise with known variance

**Comment by Balandat** [comment]
You don't need to do any masking here, you can directly use the [`FixedNoiseMultiTaskGP`](https://github.com/pytorch/botorch/blob/main/botorch/models/multitask.py#L294) model. See the docstring for the exact setting that you're describing.

**Comment by JFagin** [comment]
Thank you very much, I didn't realize that. Is there a full example for the multitask case anywhere? I don't really understand what the i1 and i2 are when making train_X. Are they just specifying the indices of X1 and X2. For example. if I had three outputs could I use X3 = torch.rand(30,2) and i3 = 2*torch.ones(30,1)? 

Also one last question how would I change the kernel? I would like to use gpytorch.kernels.MaternKernel(nu=0.5).

**Comment by saitcakmak** [comment]
Hi @JFagin. You can find a multi-task BO tutorial at https://botorch.org/tutorials/composite_mtbo. To change the kernel, you can pass `covar_module=<your kernel>` into the model constructor. By default, it uses Matern 2.5 kernel: https://github.com/pytorch/botorch/blob/main/botorch/models/multitask.py#L165-L170

**Comment by JFagin** [comment]
Thank you both so much for the help and the quick responses! I think I can take it from here.
